### PR TITLE
Replace Mutex to AtomicUsize when allocating id.

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: tests/raftstore/server.rs [hunk 1]
```diff
@@ -50,15 +51,13 @@ impl ServerCluster {
             handles: HashMap::new(),
             addrs: HashMap::new(),
             conns: Mutex::new(HashMap::new()),
-            msg_id: Mutex::new(0),
+            msg_id: AtomicUsize::new(1),
             pd_client: pd_client,
         }
     }
 
     fn alloc_msg_id(&self) -> u64 {
```
Comment (BusyJay) [inline tests/raftstore/server.rs]: Why not just use usize directly?

File: tests/raftstore/server.rs [hunk 2]
```diff
@@ -50,15 +51,13 @@ impl ServerCluster {
...
-        let mut msg_id = self.msg_id.lock().unwrap();
-        *msg_id += 1;
-        *msg_id
+        self.msg_id.fetch_add(1, Ordering::SeqCst) as u64
```
Comment (BusyJay) [inline tests/raftstore/server.rs]: I think `Ordering::Relaxed` is enough.

**Comment by BusyJay** [comment]
There are msg_id in other files too.

File: tests/raftstore/server.rs (hunk 1 above)
Comment (ngaut) [inline tests/raftstore/server.rs]: It would be a problem if we send message from 32 platform to 64 platform.

**Comment by ngaut** [comment]
PTAL

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**